### ISSUE TITLE
"plugin-desktopswitch" 

### ISSUE THREAD

**Issue by louies0623** [lxqt/lxqt-panel#2101]
**Is your feature request related to a problem? Please describe.**
![VirtualBox_PC-1_30_08_2024_21_43_12](https://github.com/user-attachments/assets/95aa5b3b-398b-4c99-8c65-6837b6173b43)

When using it, because the laptop screen is limited, I hope there is only one icon for this function. However, when I click with the mouse (or the laptop only has a Pointing stick), there is no menu that I can click on. The required desktop menu probably looks like this.

**Describe the solution you'd like**
![de](https://github.com/user-attachments/assets/d360492c-cb66-4c40-ab13-eeb95088deee)

**Comment by tsujan** [comment]
I didn't understand this report. @stefonarch, did you?

**Comment by stefonarch** [comment]
@louies0623   would like to have one small (one icon) button with a menu for the 4 workspaces instead the 1 2 3 4 button we provide. That's my understanding.

**Comment by tsujan** [comment]
But that's already implemented -- I use it under X11.

**Comment by stefonarch** [comment]
No, you can show all desktops or only the active, and you can show them on more lines like below, but no single button which expands the workspaces.

![immagine](https://github.com/user-attachments/assets/676eb572-1175-4fef-ace3-95b3abec3381)

**Comment by louies0623** [comment]
It is indeed as he said.
https://github.com/lxqt/lxqt-panel/issues/2101#issuecomment-2322550459

but the "concentrated icon" of the four grids does not look comfortable because the words are small and dense, which makes users feel uncomfortable.
![image](https://github.com/user-attachments/assets/efb1d8f8-d076-4a48-a1ba-824c41c8c245)

Just a single icon like this
![363199678-95aa5b3b-398b-4c99-8c65-6837b6173b43](https://github.com/user-attachments/assets/44aef965-4e9f-457d-b8fd-874be4237d2c)

**Comment by stefonarch** [comment]
You could use a autohiding panel on the left or right side with the switcher on it.
In the picture above with just 1 you can also switch by mousewheel workspaces.

**Comment by louies0623** [comment]
The scroll wheel is indeed an option, but what I encountered is this.
![image](https://github.com/user-attachments/assets/22246167-0996-4679-95b8-719bba51b34d)

![images - 2024-08-31T154433 685](https://github.com/user-attachments/assets/e30faf8d-9942-4487-b51c-aeb6320933f2)

**Comment by stefonarch** [comment]
Usually  WMs have a shortcut for "next workspace" so you could create a custom command  using `xdotool`to type the shortcut and display also the actual workspace. Left click → next workspace



https://github.com/user-attachments/assets/86dfc7fa-a4dc-4ea2-bbd1-ef51addefb57

**Comment by louies0623** [comment]
Just right-click to display a menu with four desktop options. Is it necessary to do such advanced operations?

**Comment by stefonarch** [comment]
Those are your options atm.

> ...until somebody picks up this request and will make a PR.
